src/models/UserMapper.php

```diff
@@ -642,4 +642,43 @@ public function resetPassword($token, $password)
 
         return false;
     }
+
+    public function delete($user_id)
+    {
+        try {
```

Comment:
Shouldn't we start a transaction before doing all those deletions? As currently the state will be somewhat undefined when an exception is thrown as we don't know at what point that exception was thrown. With a transaction the last point within the try would be a commit and within the catch we'd do a rollback and be in a known state (no user deleted)